src/arrayMethodLastIndexOf.js

```diff
@@ -5,7 +5,29 @@
  */
 function applyCustomLastIndexOf() {
   [].__proto__.lastIndexOf2 = function(searchElement, fromIndex) {
-    // write code here
+    const index = -1;
+    let lengthOfArray = this.length;
+
+    if (fromIndex < 0) {
+      lengthOfArray += fromIndex;
+    } else if (fromIndex > 0) {
+      lengthOfArray = fromIndex;
```

Comment:
you are changing the meaning of lengthOfArray here. It means something different now than array length. Rewrite the code to avoid this.